src/api.js

```diff
@@ -1362,9 +1364,12 @@ api.declare({
     res.once('close', accept);
   });
 
-  let result = await this.workClaimer.claim(
-    provisionerId, workerType, workerGroup, workerId, count, aborted,
-  );
+  let [result] = await Promise.all([
+    this.workClaimer.claim(
+      provisionerId, workerType, workerGroup, workerId, count, aborted,
+    ),
+    this.workerInfo.provisionerSeen(provisionerId),
```

Comment:
We talked a little about this in irc and decided to leave it as-is for now: running the claim operation and the provisionerSeen operation in parallel.  In most cases the provisionerSeen will just be checking a timestamp, so it will be much faster, and anyway claim is generally long-polling.